cpu/native/periph/timer.c

```diff
@@ -27,6 +27,8 @@
 #include <mach/mach_init.h>
 #include <mach/mach_port.h>
 #include <mach/mach_host.h>
+/* Both OS X and RIOT typedef thread_t. timer.c does not use either thread_t. */
+#define thread_t riot_thread_t
```

Comment:
@authmillenon defines work for latter code, not prior. The name `riot_thread_t` makes sense, because the latter declaration (our thread_t) is renamed, not the prior (OSX's thread_t).